packages/eas-cli/src/commands/secrets/create.ts

```diff
@@ -0,0 +1,153 @@
+import Log from '../../log';
+import { ensureProjectExistsAsync } from '../../project/ensureProjectExists';
+import {
+  isEasEnabledForProjectAsync,
+  warnEasUnavailable,
+} from '../../project/isEasEnabledForProject';
+import { findProjectRootAsync, getProjectAccountNameAsync } from '../../project/projectUtils';
+import { promptAsync } from '../../prompts';
+import { ensureLoggedInAsync } from '../../user/actions';
+
+export default class EnvironmentSecretCreate extends Command {
+  static hidden = true;
+  static description = 'Create an environment secret for on the current project or owner account.';
+
+  static args = [
+    {
+      name: 'target',
+      required: false,
+      description: 'Target location for the secret',
+      options: ['account', 'project'],
+    },
+    {
+      name: 'name',
+      required: false,
+      description: 'Name of the secret',
+    },
+    {
+      name: 'value',
+      required: false,
+      description: 'Value of the secret',
+    },
+  ];
+
+  async run() {
+    const user = await ensureLoggedInAsync();
+    let {
+      args: { name, value: secretValue, target },
+    } = this.parse(EnvironmentSecretCreate);
+
+    const projectDir = (await findProjectRootAsync()) ?? process.cwd();
+    const accountName = await getProjectAccountNameAsync(projectDir);
+
+    const {
+      exp: { slug },
+    } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: slug,
+    });
+
+    if (!(await isEasEnabledForProjectAsync(projectId))) {
+      warnEasUnavailable();
+      process.exitCode = 1;
+      return;
+    }
+
+    if (!target) {
+      const validationMessage = 'Secret target may not be empty.';
+
+      ({ target } = await promptAsync({
+        type: 'select',
+        name: 'target',
+        message: 'Where should this secret be used:',
+        choices: [
+          { title: 'Account-wide', value: 'account' },
+          { title: 'Project-specific', value: 'project' },
+        ],
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    if (!name) {
+      ({ name } = await promptAsync({
+        type: 'text',
+        name: 'name',
+        message: `Secret name:`,
+        validate: value => {
+          if (!value) {
+            return 'Secret name may not be empty.';
+          }
+
+          if (value.match(/^\w+$/)) {
+            return 'Names may contain only letters, numbers, and underscores.';
+          }
+
+          return true;
+        },
+      }));
+    }
+
+    if (!secretValue) {
+      const validationMessage = 'Secret value may not be empty.';
+
+      ({ secretValue } = await promptAsync({
+        type: 'text',
+        name: 'secretValue',
+        message: 'Secret value:',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    let secret;
+
+    if (target === 'project') {
+      if (!projectDir) {
```

Comment:
I think that `projectDir` cannot be falsy, see https://github.com/expo/eas-cli/pull/275/files#diff-1a72c4b45ae1df867f274f0111db129f7c4cb44e8bacabb8fba129410eb1ba7fR46